physicsnemo/models/domino/model.py

```diff
@@ -1130,42 +1141,116 @@ def calculate_solution_with_neighbors(
                     axis=-1,
                 )
 
-        for f in range(num_variables):
-            for p in range(num_sample_points):
-                if p == 0:
-                    volume_m_c = surface_mesh_centers
-                else:
-                    volume_m_c = surface_mesh_neighbors[:, :, p - 1] + 1e-6
-                    noise = surface_mesh_centers - volume_m_c
-                    dist = torch.sqrt(
-                        noise[:, :, 0:1] ** 2.0
-                        + noise[:, :, 1:2] ** 2.0
-                        + noise[:, :, 2:3] ** 2.0
+        if (
+            self.solution_calculation_mode == "one-loop"
+            or self.solution_calculation_mode == "compare"
```

Comment:
I've left a "compare" handle in the model in case anyone wants to validate these are equivalent.  But it should be removed before approval, please request that change :)